js/hatrac.js

```diff
@@ -828,13 +829,19 @@ var ERMrest = (function(module) {
         if (this.column.md5 && typeof this.column.md5 === 'object') row[this.column.md5.name] = this.hash.md5_hex;
         if (this.column.sha256 && typeof this.column.sha256 === 'object') row[this.column.sha256.name] = this.hash.sha256;
 
-        row[this.column.name].filename = this.file.name;
         row[this.column.name].size = this.file.size;
         row[this.column.name].mimetype = this.file.type;
         row[this.column.name].md5_hex = this.hash.md5_hex;
         row[this.column.name].md5_base64 = this.hash.md5_base64;
         row[this.column.name].sha256 = this.hash.sha256;
-        row[this.column.name].filename_ext = _getFilenameExtension(this.file.name, this.column.filenameExtFilter, this.column.filenameExtRegexp);
+        row[this.column.name].filename = this.file.name;
+        var filename_ext = _getFilenameExtension(this.file.name, this.column.filenameExtFilter, this.column.filenameExtRegexp);
+        row[this.column.name].filename_ext = filename_ext
+        // filename_basename is everything from the file name except the last ext
+        // For example if we have a file named "file.tar.zip"
+        //    => "file.tar" is the basename
+        //    => ".zip" is the extension
+        row[this.column.name].filename_basename = this.file.name.substring(0, this.file.name.indexOf(filename_ext));
```

Comment:
Since we know the exact length of the string that we want to remove, you should just remove that many characters from the end of the string (instead of using `indexOf`). Otherwise this might remove other parts of the filename.

Also `filename_ext` might be `null`.  So you should check for it and return the whole filename in this case.